app/src/main/java/zapsolutions/zap/fragments/SettingsFragment.java

```diff
@@ -50,12 +51,19 @@
 
     private SwitchPreference mSwHideTotalBalance;
     private ListPreference mListCurrency;
+    private Preference mAddPin;
+    private Preference mChangePin;
 
     @Override
     public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
         // Load the settings from an XML resource
         setPreferencesFromResource(R.xml.settings, rootKey);
 
+        mAddPin = findPreference("addPIN");
+        mChangePin = findPreference("changePIN");
+
+        pinOptionVisibility();
```

Comment:
If you call `pinOptionVisibility();` in `onResume()` I think we won't need it here, otherwise it will just be called twice.